internal/bminventory/inventory.go

```diff
@@ -1023,7 +1023,7 @@ func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, par
 
 		log.Infof("Re-used existing cluster <%s> image", params.ClusterID)
 		b.eventsHandler.AddEvent(ctx, params.ClusterID, nil, models.EventSeverityInfo, "Re-used existing image rather than generating a new one", time.Now())
-		return &cluster, nil
+		return b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: *cluster.ID})
```

Comment:
Still need to return nil for the error, below too

Reply:
`GetClusterInternal` returns `&cluster, nil`

Reply:
+1